modules/openideal_discussion/openideal_discussion.module

```diff
@@ -0,0 +1,81 @@
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function openideal_discussion_form_node_discussion_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Hide title for the discussion add form.
+  // @todo create the modal for Discussion edit form.
+  if (\Drupal::request()->query->get('_wrapper_format') == 'drupal_bootstrap4_modal') {
+    // @todo make a checks.
+    $title = \Drupal::request()->request->get('dialogOptions')['title'];
+    $form['title']['widget'][0]['value']['#default_value'] = t('In idea: @idea_title', ['@idea_title' => $title]);
+    $form['title']['#attributes']['class'] = 'd-none';
+    $form['body']['widget'][0]['#title_display'] = 'invisible';
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ *
+ * Allow to add comments in Idea discussion only for group member with
+ * such roles: Author, Co-author and Expert and the discussion status is open.
+ */
+function openideal_discussion_comment_create_access(AccountInterface $account, array $context, $entity_bundle) {
+  if (($node = \Drupal::routeMatch()->getParameter('node')) && $node->bundle() == 'discussion') {
+    /** @var \Drupal\openideal_idea\OpenidealHelper $helper */
+    $helper = \Drupal::service('openideal_idea.helper');
+    $member = $helper->getGroupMember($account, $node, 'idea-group_node-discussion');
+    $status = (bool) $node->field_status->value;
+    return AccessResult::forbiddenIf(($member && !$member->hasPermission('create group_node:discussion entity') || !$member || !$status));
+  }
+  return AccessResult::neutral();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ *
+ * Hide the discussion when field_visibility turned off and make it visible only
+ * for user with 'view unpublished group_node:discussion entity' permission.
+ */
+function openideal_discussion_node_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  if ($operation == 'view' && $entity->bundle() == 'discussion') {
+    /** @var \Drupal\openideal_idea\OpenidealHelper $helper */
+    $helper = \Drupal::service('openideal_idea.helper');
+    $member = $helper->getGroupMember($account, $entity, 'idea-group_node-discussion');
+    if (!$member || ($member && !$member->hasPermission('view unpublished group_node:discussion entity'))) {
+      return AccessResult::forbiddenIf(!$entity->field_visibility->value);
+    }
+  }
+  return AccessResult::neutral();
+}
+
+/**
+ * Implements hook_message_view().
+ */
+function openideal_discussion_message_view($message, $view_mode, $langcode) {
+  /** @var \Drupal\message\Entity\Message $message */
+  $message = $message['#message'];
+  if ($message->getTemplate()->id() == 'node_created'
+    && ($node = $message->field_node_reference->entity)
+    && $node->bundle() == 'discussion'
+    && $node->field_visibility->value) {
+    // @todo Implement logic.
```

Comment:
Please add more description to the todo comment. It should be clear for everyone what should be done here.

Reply:
I removed this logic totally, now we won't implement the visibility feature.